src/error.rs

```diff
@@ -85,19 +24,15 @@ pub enum TypeErrorType {
     Map,
     /// Link type.
     Link,
-    /// Key type.
-    Key(String),
-    /// Index type.
-    Index(usize),
 }
 
-impl From<Ipld> for TypeErrorType {
+impl From<Ipld> for KindErrorType {
```

Comment:
I'd consider removing these. It kind of makes sense, but it could be dangerous given automatic error conversion with `?`.

I.e., turn this into `KindErrorType::from_ipld` function.